File: llmops/api/app/api/routers/platform.py
from uuid import UUID

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.deps import get_current_account, get_db_session, get_platform_service
from app.models.account import Account
from app.schemas.platform import UpdateWechatConfigRequest, WechatConfigResponse
from app.services.platform_service import PlatformService
from app.shared.response import success_json, success_message

router = APIRouter(prefix="/apps/{app_id}/platforms", tags=["platform"])


@router.get("/wechat")
def get_wechat_config(
    app_id: UUID,
    session: Session = Depends(get_db_session),
    current_user: Account = Depends(get_current_account),
    svc: PlatformService = Depends(get_platform_service),
):
    config = svc.get_wechat_config(session, app_id, current_user)
    return success_json(WechatConfigResponse.from_config(config).model_dump())


@router.put("/wechat")
def update_wechat_config(
    app_id: UUID,
    req: UpdateWechatConfigRequest,
    session: Session = Depends(get_db_session),
    current_user: Account = Depends(get_current_account),
    svc: PlatformService = Depends(get_platform_service),
):
    svc.update_wechat_config(
        session,
        app_id,
        current_user,
        req.wechat_app_id,
        req.wechat_app_secret,
        req.wechat_token,
    )
    return success_message("Update app wechat config success")

File: llmops/ui/src/services/platform.ts
import { get, put } from '@/utils/request'
import type { GetWechatConfigResponse, UpdateWechatConfigRequest } from '@/models/platform'
import type { BaseResponse } from '@/models/base'


// 获取指定 Agent 的微信公众号发布配置信息
export const getWechatConfig = (app_id: string) => {
  return get<GetWechatConfigResponse>(`/apps/${app_id}/platforms/wechat`)
}

// 更新指定 Agent 的微信公众号发布配置
export const updateWechatConfig = (app_id: string, req: UpdateWechatConfigRequest) => {
  return put<BaseResponse<any>>(`/apps/${app_id}/platforms/wechat`, { body: req })
}
